File: aula03-unmanged-static-link/point_lib/Point.c
// Point.c : Defines the exported functions
//
#include "Point.h"
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

void Point_init(Point* p, int x, int y) {
	p->x = x;
	p->y = y;
	p->z = 0;
}

Point* Point_new(int x, int y) {
	Point* p = malloc(sizeof(Point));
	Point_init(p, x, y);
	return p;
}

void Point_delete(Point* p) {
	free(p);
}

double Point_module(Point* p) {
	return sqrt((double) p->x * p->x + p->y * p->y + p->z * p->z);
}

void Point_print(Point* p) {
	printf("Version 2 - Point (x = %d, y = %d, z = %d)\n", p->x, p->y, p->z);
}

File: aula04-hw02-optional-args/Point.java
public class Point {
	private int x;
	private int y;	

	public Point(int x, int y) {
		this.x = x;
		this.y = y;
	}

	public double distance(Point other) {
		final int dX = x - other.x;
		final int dY = y - other.y;
		return Math.sqrt(dX * dX + dY * dY);
	}

	public double distance() {
		return distance(new Point(0, 0));
	}

    @Override
    public String toString() {
        return "Point(x=" + this.x + ", y=" + this.y + ")";
    }    
}